sonic-ledd/scripts/ledd

```diff
@@ -34,133 +35,25 @@ Options:
-        log_info("Caught SIGINT - exiting...")
-        sys.exit(128 + sig)
-    elif sig == signal.SIGTERM:
-        log_info("Caught SIGTERM - exiting...")
-        sys.exit(128 + sig)
-    else:
-        log_warning("Caught unhandled signal '" + sig + "'")
-
-
-#============ Functions to load platform-specific classes ============
-
-# Returns platform and HW SKU
-def get_platform_and_hwsku():
-    try:
-        proc = subprocess.Popen([SONIC_CFGGEN, '-v', PLATFORM_KEY],
-                stdout=subprocess.PIPE,
-                shell=False,
-                stderr=subprocess.STDOUT)
-        stdout = proc.communicate()[0]
-        proc.wait()
-        platform = stdout.rstrip('\n')
-
-        proc = subprocess.Popen([SONIC_CFGGEN, '-m', MINIGRAPH_FILE, '-v', HWSKU_KEY],
-                stdout=subprocess.PIPE,
-                shell=False,
-                stderr=subprocess.STDOUT)
-        stdout = proc.communicate()[0]
-        proc.wait()
-        hwsku = stdout.rstrip('\n')
-    except OSError, e:
-        log_error("Cannot detect platform")
-        raise OSError("Cannot detect platform")
-
-    return (platform, hwsku)
-
-
-# Loads platform-specific LED control module from source
-def load_platform_led_control_module():
-    # Get platform and hwsku
-    (platform, hwsku) = get_platform_and_hwsku()
-
-    # Load platform module from source
-    platform_path = '/'.join([PLATFORM_ROOT, platform])
-    hwsku_path = '/'.join([platform_path, hwsku])
-
-    module_file_base = '/'.join([platform_path, 'plugins', LED_MODULE_NAME + '.py'])
-
-    module_file_docker = '/'.join([PLATFORM_ROOT_DOCKER, 'plugins', LED_MODULE_NAME + '.py'])
-
-    # If we can't locate a platform-specific module, exit gracefully, assuming this
-    # platform utilizes a hardware-based LED control solution
-    if os.path.isfile(module_file_base):
-        module_file = module_file_base
-    elif os.path.isfile(module_file_docker):
-        module_file = module_file_docker
-    else:
-        log_info("Failed to locate platform-specific %s module." % LED_MODULE_NAME)
-        return None
-
-    try:
-        module = imp.load_source(LED_MODULE_NAME, module_file)
-    except IOError, e:
-        log_error("Failed to load platform module '%s': %s" % (LED_MODULE_NAME, str(e)))
-        return None
-
-    log_info("Loaded module '%s'." % LED_MODULE_NAME)
-
-    try:
-        led_control_class = getattr(module, LED_CLASS_NAME)
-        led_control = led_control_class()
-    except AttributeError, e:
-        log_error("Failed to instantiate '%s' class: %s" % (LED_CLASS_NAME, str(e)))
-        return None
-
-    log_info("Instantiated class '%s.%s'." % (LED_MODULE_NAME, LED_CLASS_NAME))
-
-    return led_control
+class DaemonLedd(DaemonBase):
+    def __init__(self):
+        DaemonBase.__init__(self)
+
+    def __exit__(self):
+        DaemonBase.__exit__(self)
 
 #=============================== Main ================================
 
 def main():
```

Comment:
Suggest making this function a method of the daemon class and calling it `run()`. This function can be declared in the DaemonBase class which raises a `NotImplementedError` because it is expected each daemon which inherits the base class needs to implement its own `run()` method.

Then, the `main()` function in each daemon would resemble the following:

```
def main():
    daemon_ledd = DaemonLedd()
    if not daemon_ledd:
        print "Failed to instantiate LED daemon"
        sys.exit(1)

    daemon_ledd.run()
```

This applies to all daemons.

Reply:
Done